pyam/logger.py

```diff
@@ -18,7 +18,8 @@ def logger():
 @contextmanager
 def adjust_log_level(level='ERROR'):
     """Context manager to change log level"""
-    old_level = _LOGGER.getEffectiveLevel()
-    _LOGGER.setLevel(level)
+    _logger = logger()
```

Comment:
it should be enough to replace only the first line by `old_level = logger().getEffectiveLevel()` - this will ensure that the internal variable ´_LOGGER` is initialized later in the function (and avoids calling the `if-then` logic three times)

Reply:
I think calling `logger()` and using its result makes clearer consent that `_LOGGER` may be not initialized (= should not be used directly). Having `logger().getEffectiveLevel()`and `_LOGGER.setLevel(level)` looks messy from my point of view. But if you think your variant more correct I can change it